File: SEP3_Client/Mediator/Information/FriendPackage.cs
using SEP3_Client.Model.List.UserList;
using SEP3_Client.Model.Unit.User;

namespace SEP3_Client.Mediator.Information
{
    public class FriendPackage : InformationPackage
    {
        public FriendList SendList { get; set; }
        
        public FriendPackage(FriendList friendList,string keyword) : base(InformationType.FRIEND,keyword)
        {
            SendList = friendList.Copy();
        }

        public FriendPackage(Friend friend,string keyword) : base(InformationType.FRIEND,keyword)
        {
            SendList = new FriendList();
            SendList.AddNewFriend(friend);
        }

        public FriendPackage()
        {
            
        }
        
        public FriendList GetSendList() {
            return SendList;
        }
    }
}

File: SEP3_UserSystem/src/main/java/com/example/SEP3_UserSystem/mediator/information/FriendPackage.java
package com.example.SEP3_UserSystem.mediator.information;

import com.example.SEP3_UserSystem.model.domain.list.userList.FriendList;
import com.example.SEP3_UserSystem.model.domain.unit.user.Friend;

public class FriendPackage extends InformationPackage{
    private FriendList SendList;

    public FriendPackage(FriendList friendList,String keyword)
    {
        super(InformationType.FRIEND,keyword);
        SendList = friendList.copy();
    }

    public FriendPackage(Friend friend, String keyword)
    {
        super(InformationType.FRIEND,keyword);
        SendList = new FriendList();
        SendList.addNewFriend(friend);
    }

    public FriendList getSendList() {
        return SendList;
    }
}
